package com.lianhuabao.customer.mapper;

import com.baomidou.mybatisplus.core.mapper.BaseMapper;
import com.lianhuabao.customer.entity.SysUserRole;
import org.apache.ibatis.annotations.Delete;
import org.apache.ibatis.annotations.Mapper;
import org.apache.ibatis.annotations.Param;
import org.apache.ibatis.annotations.Select;

import java.util.List;

@Mapper
public interface SysUserRoleMapper extends BaseMapper<SysUserRole> {
    @Select("SELECT role_id FROM sys_user_role WHERE user_id = #{userId}")
    List<Long> getUserRoleIds(@Param("userId") Long userId);

    @Delete("DELETE FROM sys_user_role WHERE user_id = #{userId}")
    void deleteUserRoles(@Param("userId") Long userId);
}
